### PR TITLE
Fixed warnings generated by clang-tidy

### PR DESCRIPTION
**Pull request by Sebastian-81**

### CODE REVIEW

File: .clang-tidy [hunk 1]
```diff
@@ -12,11 +12,15 @@ Checks: '-*,
          -cppcoreguidelines-pro-type-reinterpret-cast,
          -cppcoreguidelines-pro-bounds-constant-array-index,
          llvm-*,
+         -llvm-header-guard,
```
Comment (Sebastian-81) [inline .clang-tidy:15]: doesn't work with `#pragma once`

File: .clang-tidy [hunk 2]
```diff
@@ -12,11 +12,15 @@ Checks: '-*,
...
+         -llvm-include-order,
```
Comment (Sebastian-81) [inline .clang-tidy:16]: only for llvm source code

File: .clang-tidy [hunk 3]
```diff
@@ -12,11 +12,15 @@ Checks: '-*,
...
          misc-*,
          -misc-no-recursion,
          -misc-non-private-member-variables-in-classes,
+         -misc-const-correctness,
```
Comment (Sebastian-81) [inline .clang-tidy:20]: produces many false positives, at least in version 17.0.3

File: .clang-tidy [hunk 4]
```diff
@@ -12,11 +12,15 @@ Checks: '-*,
...
          modernize-*,
          -modernize-avoid-c-arrays,
+         -modernize-use-nullptr,
```
Comment (Sebastian-81) [inline .clang-tidy:23]: crashes `clang-tidy`

Comment (Sebastian-81) [reply .clang-tidy:23]: maybe only in visual studio 2022, see [https://github.com/llvm/llvm-project/issues/63711](url)

Comment (Sebastian-81) [reply .clang-tidy]: re-enabled, to be checked in Visual Studio Code

File: .clang-tidy (hunk 3 above)
Comment (Sebastian-81) [reply .clang-tidy]: re-enabled, to be checked in versions > 17.0.3

File: .clang-tidy (hunk 2 above)
Comment (Sebastian-81) [reply .clang-tidy]: re-enabled `llvm-include-order` , to be checked in versions > 17.0.3

Comment (Sebastian-81) [reply .clang-tidy]: decision: `llvm-header-guard` is not used since we want to use `#pragma once`

File: ParallaxGen/src/main.cpp [hunk 5]
```diff
@@ -415,6 +422,6 @@ auto main(int ArgC, char **ArgV) -> int {
                      typeid(E).name(), E.what(), boost::stacktrace::to_string(boost::stacktrace::stacktrace()));
     cout << "Press ENTER to abort...";
     cin.get();
-    abort();
+    std::abort();
```
Comment (hakasapl) [inline ParallaxGen/src/main.cpp:425]: why `std::` here?

File: ParallaxGen/src/main.cpp [hunk 6]
```diff
@@ -136,10 +143,10 @@ void mainRunner(ParallaxGenCLIArgs &Args, const filesystem::path &ExePath) {
                Args.OutputDir.wstring());
 
   // Create bethesda game type object
-  BethesdaGame::GameType BGType = getGameTypeMap().at(Args.GameType);
+  BethesdaGame::GameType BGType = getGameTypeMap().at(Args.GameType); // NOLINT
```
Comment (hakasapl) [inline ParallaxGen/src/main.cpp:146]: what's the nolint for here?

File: ParallaxGenLib/include/patchers/PatcherComplexMaterial.hpp [hunk 7]
```diff
@@ -21,8 +21,8 @@ class PatcherComplexMaterial {
   ParallaxGenConfig *PGC;
   ParallaxGenD3D *PGD3D;
 
-  static std::unordered_set<std::wstring> DynCubemapBlocklist;
-  static bool DisableMLP;
+  static std::unordered_set<std::wstring> DynCubemapBlocklist; // NOLINT
+  static bool DisableMLP; // NOLINT
```
Comment (hakasapl) [inline ParallaxGenLib/include/patchers/PatcherComplexMaterial.hpp:25]: Why nolints here?

File: ParallaxGenLib/src/BethesdaDirectory.cpp [hunk 8]
```diff
@@ -465,8 +490,8 @@ auto BethesdaDirectory::getPluginLoadOrder(const bool &TrimExtension) const -> v
   return OutputLO;
 }
 
-auto BethesdaDirectory::getPathLower(const filesystem::path &Path) -> filesystem::path {
-  return filesystem::path(boost::to_lower_copy(Path.wstring()));
+auto BethesdaDirectory::getPathLower(const filesystem::path &Path)-> filesystem::path{
+    return {boost::to_lower_copy(Path.wstring())};
```
Comment (hakasapl) [inline ParallaxGenLib/src/BethesdaDirectory.cpp:494]: tab consistency here (you have 4 spaces instead of 2)

**Review by hakasapl — CHANGES_REQUESTED** [review]
minor comments

File: ParallaxGen/src/main.cpp (hunk 6 above)
Comment (Sebastian-81) [reply ParallaxGen/src/main.cpp:146]: checker thinks the variable is not initialized...

File: ParallaxGenLib/include/patchers/PatcherComplexMaterial.hpp (hunk 7 above)
Comment (Sebastian-81) [reply ParallaxGenLib/include/patchers/PatcherComplexMaterial.hpp:25]: checker does not like global variables

**Review by hakasapl — APPROVED** [review]
thanks

**Pull request merged**